packages/smart-contracts/src/contracts/ERC20FeeProxy.sol

```diff
@@ -58,4 +52,37 @@ contract ERC20FeeProxy {
       _feeAddress
     );
   }
+
+  /**
+   * @notice Call transferFrom ERC20 function and validates the return data of a ERC20 contract call.
+   * @dev This is necessary because of non-standard ERC20 tokens that don't have a return value.
+   * @return The return value of the ERC20 call, returning true for non-standard tokens
+   */
+  function safeTransferFrom(address _tokenAddress, address _to, uint256 _amount) internal returns (bool result) {
+    
+    // solium-disable-next-line security/no-low-level-calls
+    (result,) = _tokenAddress.call(abi.encodeWithSignature(
+      "transferFrom(address,address,uint256)",
+      msg.sender,
+      _to,
+      _amount
+    ));
+
+    /* solium-disable security/no-inline-assembly */
+    assembly {
+        switch returndatasize()
```

Comment:
What this assembly does is:
- Check the size of the return from the last function call.
If it's 0, it's a non-standard ERC20 because it didn't return anything.
If it's 32, it returned something and we load that value into the result variable.
Else something wen't wrong and we need to throw.